packages/mdc-tab-indicator/fading-foundation.js

```diff
@@ -0,0 +1,44 @@
+/**
+  * @license
+  * Copyright 2018 Google Inc. All Rights Reserved.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License")
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *      http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+*/
+
+import MDCTabIndicatorFoundation from './foundation';
+
+/**
+ * @extends {MDCTabIndicatorFoundation}
+ * @final
+ */
+class MDCFadingTabIndicatorFoundation extends MDCTabIndicatorFoundation {
+  handleTransitionEnd() {
+    super.handleTransitionEnd();
+    this.adapter_.removeClass(MDCTabIndicatorFoundation.cssClasses.FADING_ACTIVATE);
+    this.adapter_.removeClass(MDCTabIndicatorFoundation.cssClasses.FADING_DEACTIVATE);
+  }
+
+  activate() {
+    this.adapter_.registerEventHandler('transitionend', this.handleTransitionEnd_);
```

Comment:
Its a little confusing that you re-register every time `activate` is called.

I realized after looking at `MDCTabIndicatorFoundation` that `super. handleTransitionEnd()` de-registers the listener....but thats also confusing because now that `super. handleTransitionEnd()` has a side-effect the sub-class is depending on.

I think you can replace `super.handleTransitionEnd` with simply `this.adapter_.deregisterEventHandler('transitionend', this.handleTransitionEnd_);`. You will end up copy/pasting the same line of code into the two subclasses, but I think thats better than having side-effects.

It will also remove the need for `MDCTabIndicatorFoundation` to have a  `handleTransitionEnd()` method. `MDCTabIndicatorFoundation` will basically only have a  `computeClientRect()` method.